### PR TITLE
Fix altair examples by gutting soft_dependency hack

### PR DESCRIPTION
**Pull request by cpsievert**
Attempts to fix this problem on https://shiny.posit.co/py/docs/overview.html

Screenshot 2024-07-23 at 10 49 24 AM

### CODE REVIEW

File: docs/jupyter-widgets.qmd [hunk 1]
```diff
@@ -77,12 +76,6 @@ altair
 anywidget
 palmerpenguins
 jsonschema
```
Comment (cpsievert) [inline docs/jupyter-widgets.qmd:78]:
```suggestion
jsonschema
jsonschema_specifications
```

File: docs/overview.qmd [hunk 2]
```diff
@@ -162,12 +161,6 @@ altair
 anywidget
 palmerpenguins
 jsonschema
```
Comment (cpsievert) [inline docs/overview.qmd:163]:
```suggestion
jsonschema
jsonschema_specifications
```

File: docs/jupyter-widgets.qmd [hunk 3]
```diff
@@ -77,12 +76,7 @@ altair
...
-## file: soft_dependencies.py
-# Temporary workaround to inform shinylive of soft dependencies
-import anywidget
-import jsonschema
-import mypy_extensions
-import toolz
+jsonschema_specifications
```
Comment (cpsievert) [inline docs/jupyter-widgets.qmd:79]:
```suggestion
jsonschema-specifications
```

File: docs/overview.qmd [hunk 4]
```diff
@@ -162,12 +161,7 @@ altair
...
-## file: soft_dependencies.py
-# Temporary workaround to inform shinylive of soft dependencies
-import anywidget
-import jsonschema
-import mypy_extensions
-import toolz
+jsonschema_specifications
```
Comment (cpsievert) [inline docs/overview.qmd:164]:
```suggestion
jsonschema-specifications
```

**Comment by cpsievert** [comment]
Thanks, adding `jsonschema-specifications` to `requirements.txt` _and_ importing `import jsonschema_specifications` did the trick.

Also, just want to make sure you're aware @wch -- this "soft" import isn't needed on shinylive.io, just for the quarto shinylive extension. Is there maybe something easy we can do to make extension auto-discover soft dependencies?

**Comment by wch** [comment]
> Also, just want to make sure you're aware @wch -- this "soft" import isn't needed on shinylive.io, just for the quarto shinylive extension. Is there maybe something easy we can do to make extension auto-discover soft dependencies?


It is actually a hard dependency, but it looks like py-shinylive is not detecting the dependency. The problem is happening because the name of the package in some places is `jsonschema_specifications`, and in other places it's `jsonschema-specifications`. I'll take a look into making it detect the dependency correctly.

**Pull request merged**